apps/desktop-wallet/src/App.tsx

```diff
@@ -189,26 +194,76 @@ const App = () => {
 
               restorePendingTransactions(mempoolTxHashes, storedPendingTxs)
             })
-          dispatch(syncAddressesHistoricBalances())
+
+          dispatch(syncAddressesAlphHistoricBalances())
         }
       } else if (addressesStatus === 'initialized') {
-        if (!isTokensMetadataUninitialized && !isLoadingTokensMetadata && newUnknownTokens.length > 0) {
+        if (areVerifiedFungibleTokensInitialized && !isLoadingUnverifiedFungibleTokens && newUnknownTokens.length > 0) {
           dispatch(syncUnknownTokensInfo(newUnknownTokens))
         }
       }
     }
   }, [
     addressHashes.length,
     addressesStatus,
-    assetsInfo.status,
+    areVerifiedFungibleTokensInitialized,
     dispatch,
+    isLoadingUnverifiedFungibleTokens,
     isSyncingAddressData,
-    isLoadingTokensMetadata,
-    isTokensMetadataUninitialized,
     network.status,
     newUnknownTokens
   ])
 
+  // Fetch verified tokens from GitHub token-list and sync current and historical prices for each verified fungible
+  // token found in each address
+  useEffect(() => {
+    if (network.status === 'online' && !isLoadingVerifiedFungibleTokens) {
+      if (fungibleTokens.status === 'uninitialized') {
+        dispatch(syncVerifiedFungibleTokens())
+      } else if (verifiedFungibleTokenSymbols.uninitialized.length > 0) {
+        const symbols = verifiedFungibleTokenSymbols.uninitialized
+
+        dispatch(syncTokenCurrentPrices({ verifiedFungibleTokenSymbols: symbols, currency: settings.fiatCurrency }))
+        dispatch(syncTokenPriceHistories({ verifiedFungibleTokenSymbols: symbols, currency: settings.fiatCurrency }))
+      }
+    }
+  }, [
+    dispatch,
+    fungibleTokens.status,
+    isLoadingVerifiedFungibleTokens,
+    network.status,
+    settings.fiatCurrency,
+    verifiedFungibleTokenSymbols.uninitialized
+  ])
```

Comment:
@mvaivre I think it makes sense to only fetch prices for **verified** fungible tokens, not just tokens that have a **symbol**. Don't you agree?

Reply:
The important use-cases for this `useEffect` is that it will run:
1. when the **network status** changes to `online` (when for example the network settings change)
1. when **tokens are discovered** in addresses data that belong to the verified token list and we have not yet fetched the historic prices for
1. when the **currency settings** are changed

Reply:
See also https://github.com/alephium/alephium-frontend/pull/322/files#r1470716937